web/src/components/apps/Dashboard.jsx

```diff
@@ -164,33 +180,47 @@ class Dashboard extends Component {
       // no-op
     }
 
-    this.setState({
-      airgapUploader: new AirgapUploader(
-        true,
-        app.slug,
-        this.onDropBundle,
-        simultaneousUploads
-      ),
-    });
+    setAirgapUploader(
+      new AirgapUploader(true, app.slug, onDropBundle, simultaneousUploads)
+    );
   };
 
-  getAppDashboard = () => {
+  useEffect(() => {
+    if (app?.isAirgap && !airgapUploader) {
+      getAirgapConfig();
+    }
+
+    updateChecker.start(updateStatus, 1000);
+    getAppDashboardJob.start(getAppDashboard, 2000);
+    if (app) {
+      setWatchState(app);
+      getAppLicense(app);
+    }
+    return () => {
+      updateChecker.stop();
+      getAppDashboardJob.stop();
+      fetchAppDownstreamJob.stop();
+    };
+  }, []);
+
+  const getAppDashboard = () => {
     return new Promise((resolve, reject) => {
       // this function is in a repeating callback that terminates when
       // the promise is resolved
 
       // TODO: use react-query to refetch this instead of the custom repeater
-      if (!this.props.app) {
+      if (!app) {
         return;
       }
 
-      if (this.props.cluster?.id == "" && this.props.isHelmManaged === true) {
```

Comment:
replacing `this.props` had to be done one by one so I could check for anything like `this.props.cluster` and `this.state.cluster`